Add store tracking number copy to clipboard button

v18 opened the pull request:
Fixes https://github.com/Automattic/woocommerce-services/issues/1400

This PR adds a 'Copy to clipboard' button for tracking numbers in orders. Here's what it looks like:

<img width="979" alt="after-copy-to-clipboard-button" src="https://user-images.githubusercontent.com/11487924/41132960-1951df10-6a92-11e8-8bf9-da98b228141d.png">

The 'Label #1 printed' section looks a bit messy now. Does it need design clean up?

Here's what it looked like before the button:

<img width="970" alt="before-copy-to-clipboard-button" src="https://user-images.githubusercontent.com/11487924/41132962-1fa620f6-6a92-11e8-809e-ea219ad6c9f3.png">

Here's the button working:

![copy-to-clipboard-tracking-code](https://user-images.githubusercontent.com/11487924/41132966-25d5ee7a-6a92-11e8-838c-4f0a407e85f6.gif)

There's no real success indicator :(. **Suggestion:I** want to the button to say 'Copied!' after it was clicked, like in the edit post page:

![copy-to-clipboard-success](https://user-images.githubusercontent.com/11487924/41132979-40803e24-6a92-11e8-8900-6f2bb0b89bff.gif)

If this is cool with design, I'll add the success indicator to this PR, probably by adding it directly to the component `<ClipboardButton />` since a success indication should be a feature of that component, and folks using it shouldn't have to worry about that.

Questions / To dos
- [x] rebase (needed for the changes in #25393)
- [x] move 'Copied!' success indicator to `<ClipboardButton />`

v18 committed:
Store: Pass only needed props to TrackingLink

v18 committed:
Store: Add translate prop to LabelItem.propTypes

v18 committed:
Store: Add button to copy tracking number to clipboard

Show 'Copy to Clipboard' button after tracking number
in order activity log.

Record clicks using Tracks, with the carrier ID recorded.

jeffstieler commented:
> There's no real success indicator :(. I want to the button to say 'Copied!' after it was clicked, like in the edit post page

The example for using this component uses state to do so: https://github.com/Automattic/wp-calypso/blob/master/client/components/forms/clipboard-button/docs/example.jsx

v18 commented:
@jeffstieler yes, I saw that. I wanted to double-check that we wanted to use the same success indicator, and then I'd move that over to the `<ClipboardButton  />` component itself.

dmsnell commented on client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-item.js:
```diff
@@ -113,11 +121,20 @@ class LabelItem extends Component {
+						<ClipboardButton compact onCopy={ this.handleOnCopyClick } text={ label.tracking }>
+							{ translate( 'Copy to clipboard' ) }
+						</ClipboardButton>
+					</React.Fragment>
```
this doesn't matter much, but often we do this too

```js
import React, { Fragment } from 'react';

…

<Fragment>
	…
</Fragment>
```

v18 replied:
Thanks, that's neater :)

alisterscott commented:
This branch needs a rebase. Are we continuing with this PR?

v18 commented:
@alisterscott - yep, continuing, but it'll be a while before I get to it since I'm going to be on vacation and then we have a meetup.

v18 commented:
Did not move the copied functionality to `< ClipboardButton>` for now. Added the 'Copied!` success message directly in `< LabelItem>`. Rebased on master.

Here's what it looks like now:

![copied-to-clipboard](https://user-images.githubusercontent.com/11487924/51004654-8f06fe80-1509-11e9-90a7-8e2499c62ef0.gif)

DanReyLop commented:
It works and the code looks good. It needs a rebase before :shipit:, but after you docan go ahead, you don't need to wait another 6 months for a review (!)

Also, to back-port this code to the WP-Admin plugin, you'll need to cherry-pick it into the `feature/woocommerce-services-dependency` branch.
